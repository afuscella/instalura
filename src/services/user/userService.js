import { isStagingEnv } from '../../infra/env/isStagingEnv';
import { HttpClient } from '../../infra/http/httpClient';
import { authService } from '../auth/authService';

const BASE_URL = isStagingEnv
  // Backend DEV
  ? 'https://instalura-api-git-master-omariosouto.vercel.app'
  // Backend Production
  : 'https://instalura-api.vercel.app';

export const userService = {
  async getProfilePage(ctx) {
    const url = `${BASE_URL}/api/users/posts`;
    try {
      const token = await authService(ctx).getToken();
      const response = await HttpClient(url, {
        headers: {
          authorization: `Bearer ${token}`,
        },
      });
      return {
        user: {
          totalLikes: 100,
        },
        posts: response.data,
      };
    } catch (err) {
      console.log('error', err);
      throw new Error('Not able to retrieve user posts');
    }
  },
};
