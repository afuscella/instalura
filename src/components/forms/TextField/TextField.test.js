import React from 'react';
import user from '@testing-library/user-event';
import { render, screen } from '../../../infra/tests/testUtils';

import { TextField } from './index';

describe('<TextField />', () => {
  test('render component', () => {
    render(
      <TextField
        placeholder="test placeholder"
        name="testname"
        value="test value"
        onChange={() => { }}
      />,
    );

    expect(screen.getByRole('textbox')).toHaveValue('test value');
  });

  test('render component with alert role', () => {
    render(
      <TextField
        placeholder="test placeholder"
        name="testname"
        value="test value"
        onChange={() => { }}
        isTouched
        error="test error"
      />,
    );

    expect(screen.getByRole('alert')).toHaveTextContent('test error');
  });

  describe('when component input it is valid', () => {
    describe('and user is typing', () => {
      test('should update the value', () => {
        const onChangeMock = jest.fn();
        render(
          <TextField
            placeholder="test placeholder"
            name="testname"
            value="test value"
            onChange={onChangeMock}
            isTouched
          />,
        );

        const textFieldComponent = screen.getByPlaceholderText(/test placeholder/i);
        user.type(textFieldComponent, 'test placeholder changed');

        expect(onChangeMock).toHaveBeenCalledTimes(24);
      });
    });
  });

  describe('when component input it is not valid', () => {
    test('should display a feedback message at bottom', () => {
      render(
        <TextField
          placeholder="test placeholder"
          name="testname"
          value="test@value"
          onChange={() => { }}
          isTouched // true
          isFieldInvalid // true
          error="test error"
        />,
      );

      const textFieldComponent = screen.getByPlaceholderText(/test placeholder/i);
      expect(textFieldComponent).toHaveValue('test@value');
      expect(textFieldComponent).toMatchSnapshot();

      expect(screen.getByRole('alert')).toHaveTextContent('test error');
    });
  });
});
