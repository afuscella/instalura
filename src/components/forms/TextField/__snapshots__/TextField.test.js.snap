// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> when component input it is not valid should display a feedback message at bottom 1`] = `
.c0 {
  font-size: 16px;
  font-weight: 400;
  line-height: 1.25;
}

.c1 {
  width: 100%;
  border: 1px solid #88989E;
  padding: 12px 16px;
  outline: 0;
  border-radius: 12px;
  border-color: #dc3545;
}

.c1 + span {
  color: #dc3545;
  font-size: 11px;
}

<input
  class="c0 c1"
  name="testname"
  placeholder="test placeholder"
  role="textbox"
  type="text"
  value="test@value"
/>
`;
